### PR TITLE
MGMT-19209: Validate important fields have not changed on reinstall

### PR DESCRIPTION
**Pull request by carbonin**
Allow most of the reconfig manifest to change based on the current data in the ClusterDeployment and ImageClusterInstall, but some fields must not change between install and reinstall.

This commit validates these fields match between the manifest generated based on the new config and the manifest in the secret from the original install.

Resolves https://issues.redhat.com/browse/MGMT-19209

### CODE REVIEW

File: internal/installer/reinstall.go [hunk 1]
```diff
@@ -49,6 +53,20 @@ func (i *installer) WriteReinstallData(ctx context.Context, tmpWorkDir, isoWorkD
...
 }
 
+func validateReinstall(newConfig, secretConfig *imagebased.SeedReconfiguration) error {
+	if newConfig.BaseDomain != secretConfig.BaseDomain {
+		return fmt.Errorf("provided base domain (%s) must match previous base domain (%s)", newConfig.BaseDomain, secretConfig.BaseDomain)
+	}
+	if newConfig.ClusterName != secretConfig.ClusterName {
+		return fmt.Errorf("provided cluster name (%s) must match previous cluster name (%s)", newConfig.ClusterName, secretConfig.ClusterName)
+	}
+	if newConfig.Hostname != secretConfig.Hostname {
```
Comment (tsorya) [inline internal/installer/reinstall.go:63]: i wonder if hostname is important actually, it doesn't influence any key so why not to allow changing it?

Comment (carbonin) [reply internal/installer/reinstall.go:63]: Wouldn't we need the same hostname for the existing kubeconfig to still work?

Comment (carbonin) [reply internal/installer/reinstall.go:63]: Oh I guess that's a different dns entry. :shrug:

**Review by sakhoury — COMMENTED** [review]
looks good!

File: internal/installer/reinstall_test.go [hunk 2]
```diff
@@ -166,4 +166,26 @@ var _ = Describe("WriteReinstallData", func() {
...
+				Kubeconfig:        []byte("kubeconfig"),
+				KubeadminPassword: []byte("password"),
+				SeedReconfig:      []byte(data),
+			}
+
+			err = inst.WriteReinstallData(context.Background(), tmpWorkDir, isoWorkDir, idData)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring(errSubstring))
+		},
+		Entry("check base domain has not changed", "base_domain", "example.org", "provided base domain (example.com) must match previous base domain (example.org)"),
```
Comment (eranco74) [inline internal/installer/reinstall_test.go:188]: Add happy flow?

**Review by eranco74 — COMMENTED** [review]
/lgtm

File: internal/installer/reinstall_test.go (hunk 2 above)
Comment (carbonin) [reply internal/installer/reinstall_test.go:188]: The tests above handle the happy flow. Everything is testing `WriteReinstallData`

**Review by sakhoury — APPROVED** [review]

**Pull request merged**